cloud-controller-manager/do/loadbalancers.go

```diff
@@ -950,14 +970,22 @@ func healthCheckPort(service *v1.Service) (int, error) {
 
 	if len(ports) == 1 {
 		for _, servicePort := range service.Spec.Ports {
-			if int(servicePort.Port) == ports[0] {
+			if int(servicePort.Port) == ports[0] && servicePort.Protocol != v1.ProtocolUDP {
```

Comment:
If the protocol is UDP, then I think we'd fall through to the error returned in line 977 saying that the specified health check port was not found. This could make it harder than necessary for a user to debug what they did wrong.

I'd suggest we return a more specific, dedicated error when the protocol is UDP.

Reply:
Added!